cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/WorkflowHttpHandler.java

```diff
@@ -345,6 +346,34 @@ private WorkflowToken getWorkflowToken(String namespaceId, String appName, Strin
   }
 
   @GET
+  @Path("/apps/{app-id}/workflows/{workflow-id}/runs/{run-id}/nodestates")
+  public void getWorkflowNodeStates(HttpRequest request, HttpResponder responder,
+                                       @PathParam("namespace-id") String namespaceId,
+                                       @PathParam("app-id") String applicationId,
+                                       @PathParam("workflow-id") String workflowId,
+                                       @PathParam("run-id") String runId)
+    throws NotFoundException, DatasetManagementException {
+    ApplicationId appId = new ApplicationId(namespaceId, applicationId);
```

Comment:
I think the convention to follow with the new ``EntityId`` classes is ``Ids.namespace(namespaceId).app(applicationId)``